### PR TITLE
Update PR template

### PR DESCRIPTION
**Pull request by Gold856**
## Description

All comments are now uncommented to encourage people to actually delete them because they kept getting into commit descriptions and it's annoying. Also tried to make a more convincing argument for collecting test evidence to nudge people into testing and recording the results of that test to help with verifying PR functionality, as what the author has or has not tested is often unclear (and more often than not, edge cases were not tested, which is how #2227 was made.)

This also clarifies that the description should contain info about the events that led to a PR. What event led to that line being added? There happened to be discussion on the FRC Discord about vendordeps compiling against the WPILib kickoff release, which prompted discussion about PV's version locking implemented in #1772, but we couldn't ascertain why #1772 was made (which has since been edited to include that info) given the guarantees given by Gradle and so we had to trudge through Discord to figure out the events triggering that PR, then created #2307 in response (which also contains links and additional context for future readers.)

Finally, the line about settings compatibility has been future-proofed to always require compatibility to the previous season's releases, and to say "all" because of the events leading up to #2331. Seasons end when Champs ends.

## Meta

Merge checklist:
- [x] Pull Request title is [short, imperative summary](https://cbea.ms/git-commit/) of proposed changes
- [x] The description documents the _what_ and _why_
- [ ] If this PR changes behavior or adds a feature, user documentation is updated
- [ ] If this PR touches photon-serde, all messages have been regenerated and hashes have not changed unexpectedly
- [ ] If this PR touches configuration, this is backwards compatible with settings back to v2025.3.2
- [ ] If this PR touches pipeline settings or anything related to data exchange, the frontend typing is updated
- [ ] If this PR addresses a bug, a regression test for it is added

### CODE REVIEW

**Comment by samfreund** [comment]
Might be good to note that seasons end when champs end in the PR template.

**Comment by Gold856** [comment]
Done.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
 ## Description
 
-<!-- What changed? Why? (the code + comments should speak for itself on the "how") -->
+What changed? Why? (the code + comments should speak for itself on the "how")
 
-<!-- Fun screenshots or a cool video or something are super helpful as well. If this touches platform-specific behavior, this is where test evidence should be collected. -->
+Fun screenshots or a cool video or something are super helpful as well. If this touches core/platform-specific behavior, this is where test evidence should be collected. Test evidence is important to establish feature functionality.
```
Comment (mcm001) [inline .github/pull_request_template.md]: Should we try to word this more plainly?

```suggestion
Include fun testing screenshots or a cool video, to collect test evidence in a place where we can later reference it. Including proof this change was tested makes reviewing easier, helps us make sure we tested all our edge cases, and helps provide context for the future. 
```

Comment (Gold856) [reply .github/pull_request_template.md]: Yes.
